include/mbedtls/config.h

```diff
@@ -2586,6 +2604,22 @@
  */
 #define MBEDTLS_POLY1305_C
 
+/**
+ * \def MBEDTLS_PSA_CRYPTO_C
+ *
+ * Enable the Platform Security Architecture cryptography API.
+ *
+ * \warning This feature is experimental and available on an opt-in basis only.
+ * PSA APIs are subject to change. The implementation comes with less assurance
+ * and support than the rest of Mbed TLS.
+ *
+ * Module:  crypto/library/psa_crypto.c
+ *
+ * Requires: MBEDTLS_CTR_DRBG_C, MBEDTLS_ENTROPY_C
```

Comment:
Suggestion: we might want to mention that enabling this option has no effect unless you also use the build option `USE_CRYPTO_SUBMODULE`. Otherwise users might be surprised that they enabled this and don't get any of the promised APIs anyway.

Actually, we might even want the `USE_CRYPTO_SUBMODULE` build option and add `-DMBEDTLS_USE_CRYPTO_SUBMODULE` to `CFLAGS` and then in `check_config.h` error out if `MBEDTLS_PSA_CRYPTO_C` is defined but `MBEDTLS_USE_CRYPTO_SUBMODULE` isn't.

However, feel free to ignore that suggestion for this PR if you feel it would be feature creep - we can always add it in a subsequent PR.

Reply:
Updated to mention the build option.

Good idea about modifying CFLAGS. I think we'll make using the submodule the default option soon enough that it isn't investing too much time in this, though.